scripts/gdb/lvglgdb/lvgl.py

```diff
@@ -12,31 +13,29 @@
 g_lvgl_instance = None
 
 
-class LVList:
+class LVList(Value):
     """LVGL linked list iterator"""
 
-    def __init__(self, ll: gdb.Value, nodetype: Union[gdb.Type, str] = None):
+    def __init__(self, ll: Value, nodetype: Union[gdb.Type, str] = None):
         if not ll:
             raise ValueError("Invalid linked list")
+        super().__init__(ll)
 
-        self.ll = ll
         self.nodetype = (
             gdb.lookup_type(nodetype).pointer()
             if isinstance(nodetype, str)
             else nodetype
         )
         self.lv_ll_node_t = gdb.lookup_type("lv_ll_node_t").pointer()
-        self.current = ll["head"]
-        self._next_offset = ll["n_size"] + self.lv_ll_node_t.sizeof
-        self._prev_offset = ll["n_size"]
+        self.current = self.head
+        self._next_offset = self.n_size + self.lv_ll_node_t.sizeof
+        self._prev_offset = self.n_size
 
     def _next(self, node):
-        next = gdb.Value(int(node) + self._next_offset)
-        return next.cast(self.lv_ll_node_t.pointer()).dereference()
+        return Value(int(node) + self._next_offset).cast(self.lv_ll_node_t, ptr=True).dereference()
 
     def _prev(self, node):
-        prev = gdb.Value(int(node) + self._prev_offset)
-        return prev.cast(self.lv_ll_node_t)
+        return Value(int(node) + self._prev_offset).cast(self.lv_ll_node_t, ptr=True).dereference()
```

Comment:
This super long expression deserves two step calculation to make it better to understand.
```suggestion
        prev = Value(int(node) + self._prev_offset)
        return prev.cast(self.lv_ll_node_t, ptr=True).dereference()
```